crates/astria-sequencer-relayer/src/relayer/celestia_client/builder.rs

```diff
@@ -0,0 +1,148 @@
+use std::sync::Arc;
+
+use astria_core::generated::cosmos::{
+    base::tendermint::v1beta1::{
+        service_client::ServiceClient as NodeInfoClient,
+        GetNodeInfoRequest,
+    },
+    tx::v1beta1::service_client::ServiceClient as TxClient,
+};
+use http::Uri;
+use tendermint::account::Id as AccountId;
+use thiserror::Error;
+use tonic::transport::{
+    Channel,
+    Endpoint,
+};
+use tracing::trace;
+
+use super::{
+    super::State,
+    Bech32Address,
+    CelestiaClient,
+    CelestiaKeys,
+    GrpcResponseError,
+};
+
+/// An error when building the `CelestiaClient`.
+#[derive(Error, Clone, Debug)]
+#[non_exhaustive]
+pub(in crate::relayer) enum BuilderError {
+    /// Failed to Bech32-encode our Celestia address.
+    #[error("failed to Bech32-encode our celestia address {address}")]
+    EncodeAddress {
+        address: AccountId,
+        #[source]
+        source: Bech32EncodeError,
+    },
+    /// The celestia app responded with the given error status to a `GetNodeInfoRequest`.
+    #[error("failed to get celestia node info")]
+    FailedToGetNodeInfo(#[source] GrpcResponseError),
+    /// The node info response was empty.
+    #[error("the celestia node info response was empty")]
+    EmptyNodeInfo,
+}
+
+/// An error while encoding a Bech32 string.
+#[derive(Error, Clone, Debug)]
+#[error(transparent)]
+pub(in crate::relayer) struct Bech32EncodeError(#[from] bech32::EncodeError);
```

Comment:
It's to avoid having a third party type in our public Rust API.  That would make our API vulnerable to version changes in the dependency.  I approached the `celestia_client` module as if it were a standalone crate, since we spoke about maybe moving in that direction in the future.